Add comments

kripken opened the pull request.

sbc100 commented on src/library.js:
```diff
@@ -508,6 +508,8 @@ LibraryManager.library = {
       console.error('emscripten_realloc_buffer: Attempted to grow heap from ' + buffer.byteLength  + ' bytes to ' + size + ' bytes, but got error: ' + e);
 #endif
     }
+    // return 0  could be done here, but the caller will cast "undefined" into 0
```
How about `// implicit 0 return to save code size`

kripken commented:
PTAL @sbc100 - something went wrong with the branch before so I fixed it now. I also added another comment for something else which I was working on anyhow.

sbc100 commented on tools/system_libs.py:
```diff
@@ -1388,10 +1388,48 @@ def calculate(temp_files, in_temp, cxx, forced, stdout_=None, stderr_=None):
+  # will end up calling js_func. But the presence of a call to other_libc_func
+  # indicates that we will need libc_func linked in as well, so that is what the
+  # deps_info.json entry should contain.
+  #
```
This seems like a really long comment to have inline in the code.  I wonder if there is somewhere else we could document it?    I don't have any great ideas.  Maybe deps_info.txt and refer to that here?.. not ideal I guess

kripken replied:
Maybe we should write up a document about the entire linking process soon, after the simplifications. And this could go there?
